submissions/Ant-C-tech/friends-app/js/app.js

```diff
@@ -0,0 +1,314 @@
+    if (!IS_MUSIC_STOPPED_BY_USER) {
+        playMusic()
+    }
+    showSearchBar()
+    closeSideNav()
+
+    const checkedGenderInput = document.querySelector("input[type='radio'][name='gender']:checked")
+    let userChooseGender = (checkedGenderInput) ? [checkedGenderInput.getAttribute("data-gender")] : ['male', 'female']
+
+    const [userChooseMinAge, userChooseMaxAge] = document.getElementById('test-slider').noUiSlider.get()
+
+    //Filter
+    CURRENT_FRIENDS = FRIENDS_SOURCE.filter(friend => (
+        userChooseGender.includes(friend.gender) &&
+        friend.dob.age >= userChooseMinAge &&
+        friend.dob.age <= userChooseMaxAge))
+
+    const checkedSortParameterInput = document.querySelector("input[type='radio'][name='sort']:checked")
+    let sortParameter = (checkedSortParameterInput) ? checkedSortParameterInput.getAttribute("data-sort") : false
+
+    //Sort
+    switch (sortParameter) {
+        case '0-100':
+            CURRENT_FRIENDS.sort(byField('dob', 'age'))
+            break
+        case '100-0':
+            CURRENT_FRIENDS.sort(byField('dob', 'age')).reverse()
+            break
+        case 'name_a-z':
+            CURRENT_FRIENDS.sort(byField('name', 'first'))
+            break
+        case 'name_z-a':
+            CURRENT_FRIENDS.sort(byField('name', 'first')).reverse()
+            break
+        case 'lastName_a-z':
+            CURRENT_FRIENDS.sort(byField('name', 'last'))
+            break
+        case 'lastName_z-a':
+            CURRENT_FRIENDS.sort(byField('name', 'last')).reverse()
+            break
+
+        default:
+            break
+    }
+
+    changeContent(createFriendsScreen(CURRENT_FRIENDS), SHOW_ELEM_PRIMARY_ANIMATION, HIDE_ELEM_PRIMARY_ANIMATION)
+}
+
+function resetFilter() {
+    //In case user enter to app through sidenav (not using startButton)
+    if (!IS_MUSIC_STOPPED_BY_USER) {
+        playMusic()
+    }
+    showSearchBar()
+    closeSideNav()
+
+    CURRENT_FRIENDS = []
+    if (document.querySelector("input[type='radio'][name='gender']:checked")) {
+        document.querySelector("input[type='radio'][name='gender']:checked").checked = false
+    }
+    if (document.querySelector("input[type='radio'][name='sort']:checked")) {
+        document.querySelector("input[type='radio'][name='sort']:checked").checked = false
+    }
+    changeContent(createFriendsScreen(FRIENDS_SOURCE), SHOW_ELEM_PRIMARY_ANIMATION, HIDE_ELEM_PRIMARY_ANIMATION)
+}
+
+function closeSideNav() {
+    SEARCH_INPUT.value = ''
+    document.querySelector('.search__hint').classList.remove('active')
+}
+
+function activateSideNav() {
+    const sideNavIco = document.querySelectorAll('.sidenav')
+    M.Sidenav.init(sideNavIco, {})
+}
+
+function playMusic() {
+    APP_AUDIO.play()
+    MUSIC_BUTTON.classList.add('musicButton-active')
+    MUSIC_BUTTON.classList.add('musicButton-animation')
+    MUSIC_BUTTON.addEventListener('click', stopMusic, { once: true })
```

Comment:
Instead of adding and removing the listener, you can just add it once and then check if audio is playing - stop it and vice versa.

Please, check all the listeners you have and try to find how you can use listeners which were added only once.